### DISCUSSION TITLE
Question about how `uri` is used internally within *one* single instance.

### DISCUSSION THREAD

**Discussion by HolgerHuo** [mastodon/mastodon#17116] [Q&A] [answered]
Previous on Joinmastodon:

>https://discourse.joinmastodon.org/t/wondering-how-uri-values-are-used-inside-mastodon/3997/2

Hi! There’s one more question about `uri` i want to know about.
In a DB record, there are both `ID` and `uri` for one object. When an internal action performs, will the code look for `ID` or `uri` (like boosting or favoriting a local post)? And if I want to interact with some remote objects(like favoriting remote posts), it is the time when `uri` is called. Is that correct?

(Sorry for my poor understanding of the code here.)

**Comment by ClearlyClaire** [comment, marked as answer]
Yes, `uri` is used for remote interaction, REST API deals with local numerical ids, federation uses `uri`.

**Reply by HolgerHuo** [reply]
I see it!
Thanks a lot for your help!
